src/tables/post.rs

```diff
@@ -340,7 +342,7 @@ impl post {
     /// Versions are stored internally as fixed U16F16 numbers for ease of
     /// serialization, so this function stops you from having to mess with them.
     pub fn set_version(&mut self, version: f32) {
-        self.version = U16F16::from_num(version);
+        self.version = Version16Dot16::from_num(version)
```

Comment:
my intention with this patch was to make this type easy enough to work with that this sort of helper method would be unnecessary, and could be deleted?

Reply:
But we're still forcing the user to use custom types instead of friendly things like f32, so we still need a conversion method, right?

Reply:
Right, I meant that the `set_version` method could go away, and the user can construct a `Version16Dot16` directly if needed.